### ISSUE TITLE
[no-shadow] false positive for class static generics

### ISSUE THREAD

**Issue by vmatyi** [typescript-eslint/typescript-eslint#3312]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "no-shadow": "off",
    "@typescript-eslint/no-shadow": "warn",
  }
}
```

```TS
export class SomeClass<K> {
  static create<K>() {
    return new SomeClass<K>(); // warn: 'K' is already declared in the upper scope. eslint(@typescript-eslint/no-shadow)
  }
}
```

**Expected Result**

No ESLint errors about shadowed name.
The generic parameter of a class is not available in static functions, thus there is no real shadowing:

```TS
export class SomeClass<K> {
  static create() {
    return new SomeClass<K>();  // error: Static members cannot reference class type parameters. ts(2302)
  }
}
```

Renaming the static members <K> will reduce readability, as the function and meaning of the generic parameter is the same. 
The use case is very similar to a class member and a member function parameter, which is allowed, and rightly so:

```TS
export class OtherClass {
  name: string | undefined;
  setName(name: string) {
    this.name = name;
  }
}
```

This issue was mentioned as part of https://github.com/typescript-eslint/typescript-eslint/issues/2480 (2.), but I suspect it was overlooked.

**Actual Result**

'K' is already declared in the upper scope. eslint(@typescript-eslint/no-shadow)

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin-tslint` | `4.22.0` |
| `@typescript-eslint/parser`               | `4.22.0` |
| `TypeScript`                              | `3.9.9` |
| `ESLint`                                  | `7.24.0` |
| `node`                                    | `12.19.1` |

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/2592

Please use the search
